src/main/java/org/gradle/profiler/studio/StudioGradleClient.java

```diff
@@ -57,60 +56,32 @@ public StudioGradleClient(GradleBuildConfiguration buildConfiguration, Invocatio
         Logging.startOperation("Starting Android Studio at " + studioInstallDir);
         studioPluginServer = new Server("plugin");
         studioAgentServer = new Server("agent");
+        Server studioStartDetectorServer = new Server("studioStartDetector");
         StudioSandbox sandbox = StudioSandboxCreator.createSandbox(studioSandboxDir.map(File::toPath).orElse(null));
-        LaunchConfiguration launchConfiguration = new LauncherConfigurationParser().calculate(studioInstallDir, sandbox, studioPluginServer.getPort());
-        System.out.println();
-        System.out.println("* Java command: " + launchConfiguration.getJavaCommand());
-        System.out.println("* Classpath:");
-        for (Path entry : launchConfiguration.getClassPath()) {
-            System.out.println("  " + entry);
-        }
-        System.out.println("* System properties:");
-        for (Map.Entry<String, String> entry : launchConfiguration.getSystemProperties().entrySet()) {
-            System.out.println("  " + entry.getKey() + " -> " + entry.getValue());
-        }
-        System.out.println("* Main class: " + launchConfiguration.getMainClass());
+        LaunchConfiguration launchConfiguration = new LauncherConfigurationParser(studioInstallDir, sandbox)
+            .installStudioPlugin(studioStartDetectorServer.getPort(), studioPluginServer.getPort())
```

Comment:
Since we are always calling these two methods, why do we have the boolean flags in `LauncherConfigurationParser`?

Reply:
Actually, in tests we start a studio without a plugin or an agent. So there it comes handy that we have option to handle this